### DISCUSSION TITLE
milvus 0.10.1 How to build indexes on the CPU and perform search comparisons on the GPU

### DISCUSSION THREAD

**Discussion by Haohao966** [milvus-io/milvus#3675] [Q&A and General discussion] [answered]
in server_config.yaml  :  GPU card can be selected for the following two parameter Settings
I want to know how to set these two parameters,   can realize  build indexes on the CPU and perform search comparisons on the GPU
 search_devices:
      - gpu0
      - gpu1
build_index_devices:
      - gpu0
      - gpu1
I want to know

**Comment by shiyu22** [comment, marked as answer]
Hi, if you use CPU for indexing, it will consume a lot of CPU resources and memory. When you use GPU to search, you need to return the results to the CPU and the retrieval process will be affected, so it is recommended to use GPU for indexing.

In fact, we don't currently support the CPU indexing and GPU for retrieval you mentioned, so if you do have such a need, please submit an issue and tell us about your requirements.

**Reply by Haohao966** [reply]
milvus 0.10.1   https://www.milvus.io/cn/docs/v0.10.2/milvus_distributions-gpu.md#floating  
The notice dialog at the bottom of the page is described as follows:
 realize build indexes on the CPU and perform search comparisons on the GPU ,  or    index GPU  ,search CPU

**Reply by shiyu22** [reply]
I got it, and we actually want to say is that the data that the CPU and GPU build indexes is universal. For example, you can use the CPU to finish indexing, and then you can enable the GPU, i.e. set gpu.enable=ture, and then you can use the GPU to retrieve it!

We'll update the doc as well.
